code/scripting/api/libs/base.cpp

```diff
@@ -582,6 +582,47 @@ ADE_FUNC(getVersionString, l_Base, nullptr,
 	return ade_set_args(L, "s", str.c_str());
 }
 
+ADE_FUNC(getModTitle, l_Base, nullptr,
+         "Returns the title of the current mod as defined in game_settings.tbl. Will return an empty string if not defined.",
+         "string", "The mod title")
+{
+	auto str = Mod_title;
+	return ade_set_args(L, "s", str.c_str());
+}
+
+ADE_FUNC(getModVersion, l_Base, nullptr,
+         "Returns the version of the current mod as defined in game_settings.tbl. Will return an empty string if not defined.",
+         "string, number, number, number", "The mod title; the major, minor, patch version numbers -1 if invalid")
+{
+	auto version = Mod_version;
+
+	int major = -1;
+	int minor = -1;
+	int patch = -1;
+	
+	int i = 0;
+	auto str = Mod_version;
+	while (i <= 3) {
+		size_t pos = str.find_first_of('.');
+		if (pos != SCP_string::npos) {
+			auto ver = str.substr(0, pos);
+			if(!ver.empty() && std::find_if(ver.begin(), ver.end(), [](char c) { return !std::isdigit(c); }) == ver.end()){
+				if (major < 0) {
+					major = std::stoi(str.c_str());
+				} else if (minor < 0) {
+					minor = std::stoi(str.c_str());
+				}
+				str.erase(0, pos + 1);
+			}
```

Comment:
So because it's intended to not use the semantic version at the end of the string if it contains something else first, we should do ` else if (major < 0) { break; }`.  Also add a comment that clarifies that this is intended to keep the function returning only one or other type.  Even though it won't really optimize the code that much, it will make the code easier to follow since since we are explicitly avoiding it instead of implicitly.

I think the one thing you can improve on is more comments in general, too.  Especially when we get somewhat complicated code like this.  Not that it's impossible to read, but comments make things a lot easier.

Reply:
Done